content/sensu-go/5.15/reference/apikeys.md

```diff
@@ -0,0 +1,193 @@
+  username: user1
+  groups:
+  - group1
+  - group2
+  created_at: 1569874845
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "apikeys",
+  "api_version": "core/v2",
+  "metadata" : {
+    "name": "my-api-key"
+  },
+  "spec": {
+    "key": "key-id",
+    "username": "user1",
+    "groups": [
+      "group1",
+      "group2"
+      ]
+    "created_at": 1569874845
+    }
+  }
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+## API key specification
+
+### Top-level attributes
+
+type         | 
+-------------|------
+description  | Top-level attribute specifying the resource type. API keys should always be of type `apikeys`.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"type": "apikeys"{{< /highlight >}}
+
+api_version  | 
+-------------|------
+description  | Top-level attribute specifying the Sensu API group and version. This attribute should always be `core/v2`.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"api_version": "core/v2"{{< /highlight >}}
+
+metadata     | 
+-------------|------
+description  | Top-level collection of metadata about the API key, including the `name`. The `metadata` map is always at the top level of the check definition. This means that in `wrapped-json` and `yaml` formats, the `metadata` scope occurs outside the `spec` scope.
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}"metadata": {
+  "name": "my-api-key"
+}{{< /highlight >}}
+
+spec         | 
+-------------|------
+description  | Top-level map that includes the API key [spec attributes](#spec-attributes).
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}"spec": {
+    "key": "key-id",
+    "username": "user1",
+    "groups": [
+      "dev",
+      "ops"
+      ]
+    "created_at": 1569874845
+    }
+{{< /highlight >}}
+
+### Metadata attributes
+
+| name       |      |
+-------------|------
+description  | A unique string used to identify the API key. API key names cannot contain special characters or spaces (validated with Go regex [`\A[\w\.\-]+\z`](https://regex101.com/r/zo9mQU/2)).
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "my-api-key"{{< /highlight >}}
+
+### Spec attributes
+
+| key        |      |
+-------------|------
+description  | The API key ID.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"key": "key-id"{{< /highlight >}}
+
+| username   |     |
+-------------|------
+description  | The user associated with the API key.
+required     | true
+type         | Array
+example      | {{< highlight shell >}}"username": "user1"{{< /highlight >}}
+
+| groups     |      |
+-------------|------
+description  | The groups the user belongs to.
+required     | false
+type         | Array
+example      | {{< highlight shell >}}"groups": [
+  "group1",
+  "group2"
+]{{< /highlight >}}
```

Comment:
This field can be removed. We removed this during a later phase in the specification.